src/Razor/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDocumentClassifierPass.cs

```diff
@@ -100,7 +100,12 @@ protected override CodeTarget CreateTarget(RazorCodeDocument codeDocument, Razor
             {
                 @class.BaseType = ComponentsApi.ComponentBase.FullTypeName;
 
-                var typeParamReferences = documentNode.FindDirectiveReferences(ComponentTypeParamDirective.Directive);
+                // Constrained type parameters are only supported in Razor language versions v6.0
+                var razorLanguageVersion = codeDocument.GetParserOptions()?.Version ?? RazorLanguageVersion.Latest;
```

Comment:
Wouldn't we want to assume lowest version instead of highest if there isn't one specified?

Reply:
AFAIK, the only scenario where it is undefined is in some test cases for this pass that don't fully configure the `ComponentDocumentClassifierPass` with parser options. For that scenario, I figured it made sense to update to the latest.

Assuming this happens to a user, I think it also makes sense to default to the latest since the constrained type param directive can still support unconstrained scenarios.